source/documentation/continuous-deployment/pool-resource.md

```diff
@@ -0,0 +1,162 @@
+
+### How it works
+
+The pool-resource is backed by a git repository, where each pool is a directory
+with two subdirectories:
+
+- claimed
+- unclaimed
+
+where unclaimed locks are files inside the `unclaimed` directory, and `claimed`
+directory contains files representing claimed locks.
+
+```
+.
+├── pool-1
+│   ├── claimed
+│   └── unclaimed
+├── pool-2
+│   ├── claimed
+│   └── unclaimed
+└── pool-3
+    ├── claimed
+    │   └── f3cb3823-a45a-49e8-ab41-e43268494205
+    └── unclaimed
+```
+
+### Setting up the pool-resource
+
+You should create a pipeline to set up and manage your team's pool-resource.
+
+The following pipeline sets up three pools:
+
+- pool-1
+- pool-2
+- pool-3
+
+⚠️ Re-running the `init-pool` job will reset the state of any locks.
+
+<pre><code>---
+resources:
+  - name: pool-repo
+    type: git
+    icon: github
+    source:
+      branch: pool
+      uri: ((readonly_codecommit_pool_uri))
+      private_key: ((readonly_codecommit_private_key))
+
+jobs:
+  - name: init-pool
+    serial: true
+    plan:
+      - task: init-pool
+        config:
+          platform: linux
+          image_resource:
+            type: registry-image
+            source:
+              repository: governmentpaas/git-ssh
+              tag: latest
+          outputs:
+          - name: repo
+          run:
+            dir: repo
+            path: sh
+            args:
+            - -euo
+            - pipefail
+            - -c
+            - |
+              git init .
+              for pool in pool-1 pool-2 pool-3; do
+                mkdir -p "$pool/claimed"
+                mkdir -p "$pool/unclaimed"
+                touch "$pool/claimed/.keep"
+                touch "$pool/unclaimed/.keep"
+                git add "$pool"
+                git commit -m "setup $pool"
+              done
+
+      - put: pool-repo
+        params:
+          repository: pool
+          force: true
+</code></pre>
+
+### Using the pool-resource
+
+Once you have set up a pool-resource, you can use it to claim and release
+locks.
+
+<pre><code>---
+resource_types:
+  - name: pool
+    type: registry-image
+    source:
+      repository: concourse/pool-resource
+      tag: 1.1.3
+
+resources:
+  - name: pool
+    type: pool
+    icon: pool
+    source:
+      branch: pool
+      uri: ((readonly_codecommit_pool_uri))
+      private_key: ((readonly_codecommit_private_key))
+      pool: pool-1
+
+    - name: my-lock-config
+      type: mock
+      source:
+        create_files:
+          name: my-lock
+          metadata: ''
+
+jobs:
+  - name: add-my-lock
+    serial: true
+    plan:
+      - get: my-lock-config
+
+      - put: pool
+        params:
+          add: my-lock-config
+
+  - name: use-my-lock
+    serial: true
+    plan:
+      - put: pool
+        params:
+          claim: my-lock
+
+      - task: do-a-thing-with-my-lock
+        config: # omitted for brevity
+
+      - put: pool
+        params:
+          release: pool
```

Comment:
```suggestion
          release: my-lock
```

Reply:
> `release`: If set, we will release the lock by moving it from claimed to unclaimed. The value is the path of the lock to release (a directory containing name and metadata), which typically is just the step that provided the lock (either a get to pass one along or a put to acquire).

We do a `put` to a resource called `pool` which then does an implicit `get` into a directory called `pool` which contains a file called `name` containing `my-lock`.

Reply:
I'm not sure I understand any of that. I was merely pointing out the asymmetry between the `claim` and the `release`, which strikes me as odd. But if it's the Right Way, then fine :).

Reply:
So doing

```
- put: pool
  params:
    claim: my-lock
```

creates a directory
```
pool
  name # contents "my-lock"
  metadata # ...
```

and then doing

```
- put: pool
  params:
    release: pool
```

says, release the lock defined in the directory `pool`

---

It is not easily grokkable behaviour :/